test/integration/test_seq_copy_int.py

```diff
@@ -158,39 +171,107 @@ def test_seq_copy(train_params: str,
+        # train a minimal default model
+        data = run_train_translate(train_params=train_params,
+                                   translate_params=translate_params,
+                                   data=data,
+                                   max_seq_len=_LINE_MAX_LENGTH + C.SPACE_FOR_XOS)
+
+        _test_checkpoint_decoder(data['dev_source'], data['dev_target'], data['model'])
+        _test_parameter_averaging(data['model'])
+        _test_extract_parameters_cli(data['model'])
+        _test_evaluate_cli(data['test_outputs'], data['test_target'])
+
+
+def _test_evaluate_cli(test_outputs: List[str], test_target_path: str):
+    """
+    Runs sockeye-evaluate CLI with translations and a reference file.
+    """
+    with TemporaryDirectory(prefix="test_evaluate") as work_dir:
+        # write temporary output file
+        out_path = os.path.join(work_dir, 'hypotheses')
+        with open(out_path, 'w') as fd:
+            for output in test_outputs:
+                print(output, file=fd)
+        # Run evaluate cli
+        eval_params = "{} --hypotheses {hypotheses} --references {references} --metrics {metrics}".format(
+            sockeye.evaluate.__file__,
+            hypotheses=out_path,
+            references=test_target_path,
+            metrics="bleu chrf rouge1")
+        with patch.object(sys, "argv", eval_params.split()):
+            sockeye.evaluate.main()
+
+
+def _test_extract_parameters_cli(model_path: str):
+    """
+    Runs parameter extraction CLI and asserts that the resulting numpy serialization contains a parameter key.
+    """
+    extract_params = "--input {input} --names target_output_bias --list-all --output {output}".format(
+        output=os.path.join(model_path, "params.extracted"), input=model_path)
+    with patch.object(sys, "argv", extract_params.split()):
+        sockeye.extract_parameters.main()
+    with np.load(os.path.join(model_path, "params.extracted.npz")) as data:
+        assert "target_output_bias" in data
+
+
+def _test_parameter_averaging(model_path: str):
+    """
+    Runs parameter averaging with all available strategies
+    """
+    for strategy in C.AVERAGE_CHOICES:
+        points = sockeye.average.find_checkpoints(model_path=model_path,
+                                                  size=4,
+                                                  strategy=strategy,
+                                                  metric=C.PERPLEXITY)
+        assert len(points) > 0
+        averaged_params = sockeye.average.average(points)
+        assert averaged_params
+
+
+def _test_checkpoint_decoder(dev_source_path: str, dev_target_path: str, model_path: str):
+    """
+    Runs checkpoint decoder on 10% of the dev data and checks whether metric keys are present in the result dict.
+    """
+    with open(dev_source_path) as dev_fd:
+        num_dev_sent = sum(1 for _ in dev_fd)
+    sample_size = min(1, int(num_dev_sent * 0.1))
```

Comment:
used to be 1%. A specific reason to increase? Anyway, shouldn't really matter.

Reply:
no reason, just thought that it would be more useful to a couple through.